Add generic quad airframes for 250, 450, and 650 size quadrotors and set PWM_OUT in rc.mc_defaults.

mcsauder opened the pull request:
**Describe problem solved by the proposed pull request**
This PR adds a range of generic quadrotor airframes to advance PR #9849.

**Additional context**
See PR #9849 and PR #12506.

@dagar , would you let me know if you'd like to see anything different in this PR or if it is no longer of interest?  Thanks!

dagar commented:
From a user perspective I like this idea in theory, but I think the question is if the tuning is actually going to be valid across all the other variables (battery size, motors, ESCs, props, etc) for the same frame. 

With that in mind maybe some vaguely defined generic small, medium, large sizes as a rough starting point, but not get carried away?

@MaEtUgR @bresch what do you think?

mcsauder commented:
I've reduced the number of generic quad airframes previously added down to 250, 450, and 650 and moved the contents of `4050_generic_250` to `4250_generic_quad_250` (sans those values that are defaults).  Last, I've moved `set PWM_OUT 1234` to the `rc.mc_default` file, removed it from all of the other quadrotor airframe files that call `rc.mc_defaults`, and attended to `4013_bebop` which was the single exception requiring `set PWM_OUT none`.

Let me know if you have any other suggestions I could attend to with this PR.  Thanks!

mcsauder commented:
@dagar, @MaEtUgR , and @bresch , it fits in flash for the moment... ;)

mcsauder commented:
@dagar, if we don't go ahead with this one, what do you think about closing out this PR and #9849?

bkueng commented on ROMFS/px4fmu_common/init.d/airframes/4650_generic_quad_650:
```diff
@@ -0,0 +1,28 @@
+if [ $AUTOCNF = yes ]
+then
+	param set MC_ROLL_P 6.5
+	param set MC_ROLLRATE_P 0.05
```
That is exremely low. Did you test this?

bkueng commented on ROMFS/px4fmu_common/init.d/airframes/4650_generic_quad_650:
```diff
@@ -0,0 +1,28 @@
+
+	param set MC_PITCH_P 6.5
+	param set MC_PITCHRATE_P 0.05
+	param set MC_PITCHRATE_I 0.05
```
Please keep all I gains default (only where set higher than the default (0.2), you can keep it)

bkueng commented on ROMFS/px4fmu_common/init.d/airframes/4450_generic_quad_450:
```diff
@@ -0,0 +1,29 @@
+if [ $AUTOCNF = yes ]
+then
+	param set MC_ROLL_P 7.0
+	param set MC_ROLLRATE_P 0.15
```
These are all default, can you drop them?

mcsauder replied:
I think @dagar 's original intent here was basically a placeholder.  I don't have a 450 airframe to know if the default values are appropriate.  Do we know of any other good candidate 450 sized airframes aside from the flame wheel that might need different parameter values?

For now, I'll add a commit that drops these values and also drops the default values from `4011_dji_f450`.  Thanks @bkueng !

bkueng commented on ROMFS/px4fmu_common/init.d/airframes/4250_generic_quad_250:
```diff
@@ -35,18 +35,17 @@ then
+	# enable high-rate logging profile (helps with tuning)
+	param set SDLOG_PROFILE 19
+
+	# use thrust curve factor (instead of TPA)
```
```suggestion
	# use thrust curve factor
```

mcsauder replied:
Done. Thanks!

bkueng commented on ROMFS/px4fmu_common/init.d/airframes/4250_generic_quad_250:
```diff
@@ -1,32 +1,32 @@
 #!/bin/sh
 #
-# @name Generic 250 Racer
+# @name Generic Quadcopter (250mm)
```
This is specifically for racers, can you keep it in the name?

mcsauder replied:
Done. Thanks!

mcsauder commented on ROMFS/px4fmu_common/init.d/airframes/4011_dji_f450:
```diff
@@ -10,24 +10,10 @@
-	param set MC_YAWRATE_P 0.3
-	param set MC_YAWRATE_I 0.1
-	param set MC_YAWRATE_D 0
-
```
For explanation of these deletions, all of these are all default parameter values.

mcsauder commented:
Closing this; it can be revisited in the future.  Thanks for your reviews!
